File: 599-minimum-index-sum-of-two-lists/599-minimum-index-sum-of-two-lists.js
/**
 * @param {string[]} list1
 * @param {string[]} list2
 * @return {string[]}
 */
var findRestaurant = function(list1, list2) {
    let result = []
    let map = new Map()
    for (let i = 0; i < list1.length; i++) {
        map.set(list1[i], i)
    }
    let min = Infinity
    for (let i = 0; i < list2.length; i++) {
        if (map.has(list2[i])) {
            if (map.get(list2[i]) + i < min) {
                result = [list2[i]]
                min = map.get(list2[i]) + i
            } else if (map.get(list2[i]) + i === min) {
                result.push(list2[i])
			}
		}
    }
    return result
};

File: 599-minimum-index-sum-of-two-lists/599-minimum-index-sum-of-two-lists.cs
public class Solution {
    public string[] FindRestaurant(string[] list1, string[] list2) {
        List<string> result = new List<string>();
        
        Dictionary<string, int> dict = new Dictionary<string, int>();
        
        for(int i = 0; i < list1.Length; i++) {
            dict.Add(list1[i], i);
        }
        
        int min = int.MaxValue;
        
        int sum;
        
        for(int i = 0; i < list2.Length; i++) {
            if(dict.ContainsKey(list2[i])) {
                sum = dict[list2[i]] + i;
                if(sum < min) {
                    result = new List<string>();
                    result.Add(list2[i]);
                    min = sum;
                } else if (sum == min) {
                    result.Add(list2[i]);
                }
            }
        }
        
        return result.ToArray();
    }
}